src/ConditionalTokensMapping.ts

```diff
@@ -102,11 +110,18 @@ export function handlePositionsMerge(event: PositionsMerge): void {
 
   // If the user has merged a full set of outcome tokens then we want to update their market position accordingly
   if (partitionCheck(merge.partition, condition.outcomeSlotCount)) {
-    // This is not ideal as in theory we could have multiple market makers for the same condition
-    // Given that this subgraph only tracks market makers deployed by Polymarket, this is acceptable for now
-    let marketMakers = condition.fixedProductMarketMakers;
-    for (let i = 0; i < marketMakers.length; i += 1) {
-      updateMarketPositionsFromMerge(marketMakers[i], event);
+    // // This is not ideal as in theory we could have multiple market makers for the same condition
+    // // Given that this subgraph only tracks market makers deployed by Polymarket, this is acceptable for now
+    // let marketMakers = condition.fixedProductMarketMakers;
+    // for (let i = 0; i < marketMakers.length; i += 1) {
+    //   updateMarketPositionsFromMerge(marketMakers[i], event);
+    // }
+    // We explicitly only want to track one FPMM per condition
+    if (condition.fixedProductMarketMakers.length > 0) {
+      updateMarketPositionsFromMerge(
+        condition.fixedProductMarketMakers[0],
```

Comment:
@JonathanAmenechi How do we know this is the fpmm that corresponds to USDC and the fpmm we want?

Reply:
This is a fair question but it is pretty trivial to add a check for USDC only FPMMs here.


Important to note that it is possible for multiple FPMMs to be created for the same condition and collateral(see [1](https://polygonscan.com/address/0x1a7054cd7b650a56cfa6974494e0d9014852edc1) and [2](https://polygonscan.com/address/0x4fb01253e61e1ed69e7a5bacdd444e59a1cc2bb7)).


In such a case, it's tough to know which FPMM to use.